File: timeline-service/src/main/java/com/cardinalis/timelineservice/model/Timeline.java
package com.cardinalis.timelineservice.model;

import lombok.*;
import org.cardinalis.tweetservice.model.Tweet;
import org.springframework.data.redis.core.RedisHash;

import javax.persistence.Id;
import java.io.Serializable;
import java.util.List;

@Data
@RedisHash("TIMELINE")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Timeline implements Serializable {
    private static final long serialVersionUID = 10L;

//    @Id
//    private String username;

    private List<Tweet> userTimeline;
}

File: timeline-service/src/main/java/com/cardinalis/timelineservice/model/Timeline.kt
package com.cardinalis.timelineservice.model;

import org.cardinalis.tweetservice.model.Tweet
import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.index.Indexed

@RedisHash("Timeline")
data class Timeline(
        @Indexed @Id var userId: String,
//        var items: MutableList<Tweet>? = mutableListOf()
) {}